rdt/hyper_transformer.py

```diff
@@ -180,26 +175,12 @@ def transform(self, data):
         if self.copy:
             data = data.copy()
 
-        drop_columns = []
-        for column_name, transformer in self._transformers.items():
-            if column_name in data:
-                column = data[column_name]
-                transformed = transformer.transform(column)
-
-                shape = transformed.shape
-
-                if len(shape) == 2:
-                    drop_columns.append(column_name)
-                    for index in range(shape[1]):
-                        new_column = '{}#{}'.format(column_name, index)
-                        data[new_column] = transformed[:, index]
-
-                else:
-                    data[column_name] = transformed
-
-        if drop_columns:
-            data.drop(drop_columns, axis=1, inplace=True)
+        for transformer in self._transformers_sequence:
+            data = transformer.transform(data, drop=False)
 
+        columns_to_drop = [col for col in data
```

Comment:
nit: I'd use hanging indent here, in three lines (output elements, for loop, condition), and also call it column

```python3
columns_to_drop = [
    column
    for column in data
    if column in self._input_columns or column in self._generated_columns
]